### DISCUSSION TITLE
P12 file with password

### DISCUSSION THREAD

**Discussion by zeehan805** [CodeDredd/laravel-soap#250] [Q&A]
How I can add p12 file pfx and other certificate in request

**Comment by veewee** [comment]
p12 is currently not supported in wsse-middleware.
You can convert it to x509 with following commands:

```
openssl pkcs12 -in your.p12 -out security_token.pub -clcerts -nokeys
openssl pkcs12 -in your.p12 -out security_token.priv -nocerts -nodes
```

That way, you can use them like this:

```php
$response = Soap::baseWsdl('http://test.com'/v1?wsdl)->withWsse([
    'privateKeyFile' => 'path/to/security_token.priv',
    'publicKeyFile' => 'path/to/security_token.pub',
])->call(...);
```

More info:
https://github.com/php-soap/psr18-wsse-middleware#signing-a-soap-request-with-pkcs12-or-x509-certificate

**Comment by zeehan805** [comment]
thank you for the reply what will the key name like now privateKeyFile then
I have to write other key like p12certificate?? or instead of private key
file give this certificate i hope you understand what i mean
